controllers/disruption_controller.go

```diff
@@ -533,7 +524,7 @@ func (r *DisruptionReconciler) handleChaosPodsTermination(instance *chaosv1beta1
 		if removeFinalizer || ignoreStatus {
 			r.log.Infow("chaos pod completed, removing finalizer", "target", target, "chaosPod", chaosPod.Name)
 
-			controllerutil.RemoveFinalizer(&chaosPod, chaosPodFinalizer)
+			controllerutil.RemoveFinalizer(&chaosPod, chaostypes.ChaosPodFinalizer)
```

Comment:
Are you still doing leaving this line in for the case where the pod didn't clean itself up?

Reply:
Essentially, yes. When would that be? I can't really think of a time. However, since we got here, we're "sure" this is safe, so why not do or try it? E.g., maybe somebody sigkilled a chaos pod, but its target is gone now, so we're sure it's safe to clean up. 

I wanted to supplement this check, not replace it.

Reply:
And if the finalizer was removed by the pod already, most of the time it won't even be picked up by this loop.

I follow up with this unlikely scenario:

- If unless it wasn't cleaned up by the time this loop hits, so it was erroneously picked up despite its finalizer already having been removed by the new injector code, then would this code path gracefully give up?
- How about the other direction? Could this code path start removing the pod's finalizer before the new injector code removes the finalizer? Would that cause a bad outcome on the injector code side?

Reply:
So in both cases, I would expect to get error logs (from whichever of the two updated second), but that we will eventually, successfully remove the finalizer in the controller on a later reconciliation. I will give some thought to how we can log at WARN instead, but without running into a scenario where we never actually error and leave it stuck

Reply:
Okay sounds good.

Reply:
Yes I think it is a good thing to garbage collect chaos pods in case they couldn't remove their own finalizer which can happen in a lot of cases (a pending chaos pod, a sigkilled chaos pod as Philip said, etc.). We could eventually simplify the conditions above this code by removing the `PodSucceeded` case so we do not have a collision with what's happening in the chaos pod (since it removes its finalizer only when it's succeeded anyway).